src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Generated/Recipe.cs

```diff
@@ -330,18 +332,94 @@ private void ConfigureBeanstalkEnvironment(Configuration settings)
             {
-                EnvironmentName = settings.BeanstalkEnvironment.EnvironmentName,
-                ApplicationName = settings.BeanstalkApplication.ApplicationName,
-                PlatformArn = settings.ElasticBeanstalkPlatformArn,
-                OptionSettings = optionSettingProperties.ToArray(),
-                CnamePrefix = !string.IsNullOrEmpty(settings.CNamePrefix) ? settings.CNamePrefix : null,
-                // This line is critical - reference the label created in this same stack
-                VersionLabel = ApplicationVersion.Ref,
+                OnEventHandler = GetOnEventHandler(),
+                IsCompleteHandler = GetIsCompleteHandler()
+            });
+
+            new CustomResource(this, "CustomResource", new CustomResourceProps
+            {
+                ServiceToken = customResourceProvider.ServiceToken,
+                Properties = new Dictionary<string, object>()
+                {
+                    { "ApplicationName", settings.BeanstalkApplication?.ApplicationName ?? string.Empty },
+                    { "EnvironmentName", settings.BeanstalkEnvironment?.EnvironmentName ?? string.Empty },
+                    { "VersionLabel", ApplicationVersion?.Ref ?? string.Empty }
+                }
+            });
+        }
+
+        private IFunction GetOnEventHandler()
+        {
+            IFunction onEventlambdaFunction = new Function(this, "OnEventlambdaFunction", new FunctionProps
+            {
+                Runtime = Runtime.PYTHON_3_9, // execution environment
+                Code = Code.FromAsset("Generated/UpdateBeanstalkEnvironment/OnEventLambda"), // Code loaded from the disk path
+                Handler = "OnEvent.handler", // file is "OnEvent.py", function is "handler"
+                Role = GetOnEventLambdaRole(),
+                Timeout = Duration.Minutes(5)
+            });
+
+            return onEventlambdaFunction;
+        }
+
+        private IFunction GetIsCompleteHandler()
+        {
+            IFunction isCompletelambdaFunction = new Function(this, "IsCompletelambdaFunction", new FunctionProps
+            {
+                Runtime = Runtime.PYTHON_3_9, // execution environment
+                Code = Code.FromAsset("Generated/UpdateBeanstalkEnvironment/IsCompleteLambda"), // Code loaded from the disk path
+                Handler = "IsComplete.handler", // file is "IsComplete.py", function is "handler"
+                Role = GetIsCompleteLambdaRole(),
+                Timeout = Duration.Minutes(5)
+            });
+
+            return isCompletelambdaFunction;
+        }
+
+        private IRole GetOnEventLambdaRole()
+        {
+            return new Role(this, "OnEventlambdaExecutionRole", InvokeCustomizeCDKPropsEvent("OnEventlambdaExecutionRole", this, new RoleProps
+            {
+                AssumedBy = new ServicePrincipal("lambda.amazonaws.com"),
+                ManagedPolicies = new[]
```

Comment:
Lock down the role to just the SDK call being made for the environment being updated.

Reply:
Fixed